streams/src/main/java/org/apache/kafka/streams/kstream/internals/TimeWindowedKStreamImpl.java

```diff
@@ -63,42 +63,33 @@
 
     @Override
     public KTable<Windowed<K>, Long> count() {
-        return doAggregate(
-                aggregateBuilder.countInitializer,
-                aggregateBuilder.countAggregator,
-                Serdes.Long());
+        return count(Materialized.<K, Long, WindowStore<Bytes, byte[]>>with(keySerde, Serdes.Long()));
     }
 
+    @SuppressWarnings("unchecked")
     @Override
     public KTable<Windowed<K>, Long> count(final Materialized<K, Long, WindowStore<Bytes, byte[]>> materialized) {
         Objects.requireNonNull(materialized, "materialized can't be null");
         final MaterializedInternal<K, Long, WindowStore<Bytes, byte[]>> materializedInternal
                 = new MaterializedInternal<>(materialized, builder, AGGREGATE_NAME);
+        if (materializedInternal.keySerde() == null) {
+            materializedInternal.withKeySerde(keySerde);
+        }
         if (materializedInternal.valueSerde() == null) {
-            materialized.withValueSerde(Serdes.Long());
+            materializedInternal.withValueSerde(Serdes.Long());
         }
-        return aggregate(aggregateBuilder.countInitializer, aggregateBuilder.countAggregator, materialized);
+
+        return (KTable<Windowed<K>, Long>) aggregateBuilder.build(new KStreamWindowAggregate<>(windows, materializedInternal.storeName(), aggregateBuilder.countInitializer, aggregateBuilder.countAggregator),
+                AGGREGATE_NAME,
+                materialize(materializedInternal),
+                materializedInternal.isQueryable());
     }
 
 
-    @SuppressWarnings("unchecked")
     @Override
     public <VR> KTable<Windowed<K>, VR> aggregate(final Initializer<VR> initializer,
                                                   final Aggregator<? super K, ? super V, VR> aggregator) {
-        Objects.requireNonNull(initializer, "initializer can't be null");
-        Objects.requireNonNull(aggregator, "aggregator can't be null");
-        return doAggregate(initializer, aggregator, (Serde<VR>) valSerde);
-    }
-
-    @SuppressWarnings("unchecked")
-    private <VR> KTable<Windowed<K>, VR> doAggregate(final Initializer<VR> initializer,
-                                                     final Aggregator<? super K, ? super V, VR> aggregator,
-                                                     final Serde<VR> serde) {
-        final String storeName = builder.newStoreName(AGGREGATE_NAME);
-        return (KTable<Windowed<K>, VR>) aggregateBuilder.build(new KStreamWindowAggregate<>(windows, storeName, initializer, aggregator),
-                                                                AGGREGATE_NAME,
-                                                                windowStoreBuilder(storeName, serde),
-                                                                false);
+        return aggregate(initializer, aggregator, Materialized.<K, VR, WindowStore<Bytes, byte[]>>with(keySerde, null));
```

Comment:
This is a meta explanation about the serde inheritance across multiple internal impl classes:

1. reduce: inherit the key and value serdes from the parent XXImpl class.
2. count: inherit the key serdes, enforce setting the `Serdes.Long()` for value serdes.
3. aggregate: inherit the key serdes, do not set for value serdes internally (line 92 here is for this case).